src/safeposix/syscalls/net_calls.rs

```diff
@@ -1234,4 +1233,274 @@ impl Cage {
+                if sockfdobj.localaddr == None {
+                    
+                    //sets the address to 0.0.0.0 if the address is not initialized yet
+                    //setting the family as well based on the domain
+                    let addr = match sockfdobj.domain {
+                        AF_INET => { interface::GenIpaddr::V4(interface::V4Addr::default()) }
+                        AF_INET6 => { interface::GenIpaddr::V6(interface::V6Addr::default()) }
+                        _ => { panic!("Unknown domain set"); }
+                    };
+                    ret_addr.set_addr(addr);
+                    ret_addr.set_port(0);
+                    ret_addr.set_family(sockfdobj.domain as u16);
+                    return 0;
+                }
+                
+                //if the socket is not none, then return the socket
+                *ret_addr = sockfdobj.localaddr.unwrap();
+                return 0;
+
+            } else {
+                return syscall_error(Errno::ENOTSOCK, "getsockname", "the provided file is not a socket");
+            }
+        } else {
+            return syscall_error(Errno::EBADF, "getsockname", "the provided file descriptor is not valid");
+        }
+    }
+
+    //we only return the default host name because we do not allow for the user to change the host name right now
+    pub fn gethostname(&self, length: usize, address_ptr: &mut [u8]) -> i32 {
+        if length < 0 {
+            return syscall_error(Errno::EINVAL, "gethostname", "invalid argument");
+        }
+
+        let name_length: usize = DEFAULT_HOSTNAME.chars().count();
+        if name_length > length {
+            address_ptr[..length].copy_from_slice(&DEFAULT_HOSTNAME[..length].as_bytes());
+        } else {
+            address_ptr[..name_length].copy_from_slice(&DEFAULT_HOSTNAME.as_bytes());
+        }
+        return 0;
+    }
+
+    pub fn poll_syscall(&self, fds: &mut [PollStruct], timeout: Option<interface::RustDuration>) -> i32 { //timeout is supposed to be in milliseconds
+
+        let mut return_code: i32 = 0;
+        let start_time = interface::starttimer();
+
+        let end_time = match timeout {
+            Some(time) => time,
+            None => interface::RustDuration::MAX
+        };
+
+        loop {
+            for structpoll in &mut *fds {
+                let fd = structpoll.fd;
+                let events = structpoll.events;
+
+                let read = if events & POLLIN > 0 {true} else {false};
+                let write = if events & POLLOUT > 0 {true} else {false};
+                let err = if events & POLLERR > 0 {true} else {false};
+
+                let mut reads = interface::RustHashSet::<i32>::new();
+                let mut writes = interface::RustHashSet::<i32>::new();
+                let mut errors = interface::RustHashSet::<i32>::new();
+
+                let mut mask: u32 = 0;
+
+                //passing None does the same thing as passing 0 in native Linux
```

Comment:
This is incorrect. Passing None does the same thing as passing NULL in native which according to the man page for select makes it so that it can block indefinitely. What you want to do is pass a duration of 0, which is equivalent to passing a timespec with all fields zeroed in linux.

Reply:
I think that I fixed this. Please check it out, though.